### PR TITLE
fix default value on back button press

### PR DESCRIPTION
**Pull request by slavik-lvovsky**

### CODE REVIEW

File: src/Form.vue [hunk 1]
```diff
@@ -337,7 +337,10 @@ export default {
             } catch(e) {
               this.console.error(`Could not evaluate when() for ${question.name}`);
             }
-          }
+          } else if (question.when !== false) {
+	      question.shouldShow = true;
+	      shouldValidate = true;
+	  }
```
Comment (tomer-epstein) [inline src/Form.vue:343]: 👍

File: src/Form.vue [hunk 2]
```diff
@@ -448,12 +451,20 @@ export default {
...
+	     } else {
+		_default = question.__origAnswer; 
+	     }
+	  }
+	  this.$set(question, "_default", _default);
+	}
+
+	if (question._default === undefined) {
+	   this.$set(question, "_default", question.__origAnswer);
+	}
```
Comment (tomer-epstein) [inline src/Form.vue:467]: what if used from other consumer then yo-ui and question.__origAnswer is undefined?

Comment (tomer-epstein) [reply src/Form.vue:467]: did u mean ? -> if (question.default === undefined) {

File: src/Form.vue [hunk 3]
```diff
@@ -514,11 +525,15 @@ export default {
...
-              answers[question.name] = question.answer;
+		if (question.__origAnswer === undefined) {
+			question._default = await question.default(answers);
+		} else {
+			question._default = question.__origAnswer; 
+		}
+		question.answer = this.getInitialAnswer(question);
+
+		// optimization: avoid repeatedly calling this.getAnswers()
+		answers[question.name] = question.answer;
```
Comment (tomer-epstein) [inline src/Form.vue:536]: did you verify that appltDefaultWhenDirty flag still works?

Comment (slavik-lvovsky) [reply src/Form.vue:536]: yes it works

File: src/Form.vue (hunk 2 above)
Comment (slavik-lvovsky) [reply src/Form.vue:467]: no i meant if (question._default === undefined) {
if __origAnswer is undefined than _default will also be undefined

File: src/Form.vue [hunk 4]
```diff
@@ -448,12 +448,20 @@ export default {
...
+			} else {
+				_default = question.__origAnswer; 
+			}
+		}
+		this.$set(question, "_default", _default);
+	}
+
+	if (question._default === undefined) {
+		this.$set(question, "_default", question.__origAnswer);
+	}
```
[the remark this answers is not in the document]
Comment (slavik-lvovsky) [reply src/Form.vue]: for example question with type === "input" has no default value or default function but we still need to display it's value on back

File: src/Form.vue [hunk 5]
```diff
@@ -448,12 +451,20 @@ export default {
...
+	     } else {
+		_default = question.__origAnswer; 
+	     }
+	  }
+	  this.$set(question, "_default", _default);
+	}
+
+	if (question._default === undefined) {
+	   this.$set(question, "_default", question.__origAnswer);
+	}
```
Comment (tomer-epstein) [inline src/Form.vue:467]: } else {
	if (question._default === undefined) {
	   this.$set(question, "_default", question.__origAnswer);
	}
}

Comment (slavik-lvovsky) [reply src/Form.vue]: Done

**Review by tomer-epstein — APPROVED** [review]

**Pull request merged**